client/blocks/image-selector/dropzone.jsx

```diff
@@ -0,0 +1,109 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { head, uniqueId } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DropZone from 'components/drop-zone';
+import MediaActions from 'lib/media/actions';
+import MediaStore from 'lib/media/store';
```

Comment:
Have you looked at using the newer methods to access the media data, like `redux` or the data layer? I think #26168 has some good context on this.

Reply:
I did look into this, but I'm a bit confused by how it's supposed to be implemented (or if parts are ready to be implemented).

The QueryMedia component only accepts a single `mediaId` as a prop.  It seems like it may be able to handle multiple images if passed through `query`, but I can't find any other reference to this or how to form the query.  Or is the best method here to form individual queries for each of the images?

For the new data layer, it looks like there's currently not a way to handle uploading files, so I'm not sure if we can replace the current Flux-based MediaActions implementation.

Reply:
We will be replacing all Flux implementations at some point and I’d rather us not introduce new store uses. Pinging @Automattic/team-calypso to see if there are obvious next steps we're missing.

Reply:
@nb Understood.  I'll dig around and see if I can't find more information on this while we're waiting for a response.

Reply:
I chatted with @jsnajdr about the migration of the media store from flux to redux today. It is slated to be updated ( p3fqKv-6uU-p2 ) but no ETA and it is one of the more complex stores. So I guess the question here is are we okay adding in additional usage of the current flux stores which will have to be migrated during the update?

Reply:
@timmyc When I chatted with @nb I think we got the okay to go ahead and merge since it looked like the Redux implementation for media actions needs more time but agreed let's wait for @nb to double confirm that.

We can always hold off on this merge and rebase once the Redux stores are ready.

Reply:
Let’s merge this if you don't have any blockers, @timmyc – I’d rather we get a better image selector to users sooner than wait on reduxification.